metrics/cli/docs/features/doctor_output_improvements/01_doctor_output_improvements.md

```diff
@@ -118,3 +143,200 @@ The validation output classes and models are a part of the Metrics project and s
+
+We need to update the `.checkVersions()` method of the `Doctor` class to return the [`ValidationResult`](#validationresult) that will be printed by the `DoctorCommand`.
+
+The `ValidationResult` contains the [`TargetValidationResult`s](#targetvalidationresult) for each 3-rd party service. In order to create the `TargetValidationResult`s, the `Doctor` needs to know the recommended versions and installation URLs of the services. To provide this information, we need to inject the [`Dependencies`](#dependencies) instance containing the corresponding [`Dependency`](#dependency) model for each service.
+
+##### DoctorFactory
+
+The `DoctorFactory` is a factory class responsible for the `Doctor` instance creation. We need to inject the [`DependenciesFactory`](#dependenciesfactory-and-dependenciesconstants) to the `DoctorFactory`, so that the `DoctorFactory` can obtain the instance of `Dependencies` to be used during the `Doctor` instance creation.
+
+##### CoolService and CoolServiceCli
+Assume a `CoolService` as a 3-rd party service for which we want to improve `doctor` command output.
+
+Currently, the `CoolService` has the `.version()` method that is implemented in the `CoolCliServiceAdapter`. In the current implementation, the `.version()` method runs the `version` command of some service cli and prints the output directly to the stdout.    
+
+To improve the `doctor` command output, we want to receive the `version` command output first, and then prettify it depending on the output. To handle that, we need to update the `.version()` methods of the following classes to return the `Future<ProcessResult>`: 
+1. `InfoService`;
+2. `CoolService`; 
+3. `CoolServiceCli`;
+4. `CoolCliServiceAdapter`.
+
+Consider the following class diagram that describes the structure of the updated `Metrics CLI` package:
+![Metrics CLI class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/doctor_output_design/metrics/cli/docs/features/doctor_output_improvements/diagrams/metrics_cli_class_diagram.puml)
+
+
+##### Making things work <a href="#doctor-making-things-work" id="doctor-making-things-work"></a>
+
+Consider the following steps needed to be able to improve the doctor command output:
+1. Create the `dependencies` package containing the following classes: `DependenciesConstants`, `Dependency`, `Dependencies`, `DependenciesFactory`.
+2. Update the `DoctorFactory` class to contain the instance of `DependenciesFactory`.
+3. For each service cli (e.g., `FlutterCli`, `GitCli`, etc.) update the `.version()` method to return the `ProcessResult`.
+4. For each cli service adapter (e.g., `FlutterCliServiceAdapter`, `GitCliServiceAdapter`, etc.) update the `.version()` methods to return the `ProcessResult`.
```

Comment:
:pencil:
https://developers.google.com/tech-writing/one/lists-and-tables#keep_list_items_parallel
```suggestion
3. Update the `.version()` method to return the `ProcessResult` for each service CLI (e.g., `FlutterCli`, `GitCli`, etc.)
4. Update the `.version()` methods to return the `ProcessResult` for each CLI service adapter (e.g., `FlutterCliServiceAdapter`, `GitCliServiceAdapter`, etc.)
```